pkg/csi/service/vanilla/controller.go

```diff
@@ -1592,6 +1592,59 @@ func (c *controller) calculateAccessibleTopologiesForDatastore(ctx context.Conte
 	return datastoreAccessibleTopology, nil
 }
 
+// createFileVolumeWithMultiVc creates file volumes in a multi VC setup.
+func (c *controller) createFileVolumeWithMultiVc(ctx context.Context, req *csi.CreateVolumeRequest,
+	createVolumeSpec common.CreateVolumeSpec, filterSuspendedDatastores bool) (
+	*csi.CreateVolumeResponse, string, error) {
+	log := logger.GetLogger(ctx)
+	for vcHost := range c.managers.VcenterConfigs {
+		fsEnabledClusterToDsInfoMap := c.authMgrs[vcHost].GetFsEnabledClusterToDsMap(ctx)
+
+		var filteredDatastores []*cnsvsphere.DatastoreInfo
+		for _, datastores := range fsEnabledClusterToDsInfoMap {
+			filteredDatastores = append(filteredDatastores, datastores...)
+		}
+
+		if len(filteredDatastores) == 0 {
+			log.Errorf("no datastores found to create file volume, vsan file service may be disabled on VC %s", vcHost)
+			continue
+		}
+
+		vc, err := c.managers.VcenterManager.GetVirtualCenter(ctx, vcHost)
+		if err != nil {
+			log.Errorf("Failed to get VC for vcHost %s, err: %+v", vcHost, err)
+			continue
+		}
+		volumeID, faultType, err := common.CreateFileVolumeUtil(ctx, cnstypes.CnsClusterFlavorVanilla,
```

Comment:
Don't we need to consider topology requirement came along with the CreateVolume request?

Reply:
Addressed in the latest commit.